packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx

```diff
@@ -81,8 +110,38 @@ it('renders panel with a show-on-hover menu icon if prop menu', () => {
   expect(screen.getByTestId('panel-menu-button')).not.toBeVisible();
 });
 
-it.skip('renders states in the panel header if any given', () => {});
+it('renders error status in the panel header if any given', () => {
+  setup({ statusMessage: 'Error test' });
+
+  expect(screen.getByTestId('panel-status-wrapper')).toBeInTheDocument();
+});
+
+it('does not render error status in the panel header if loadingState is error, but no statusMessage', () => {
```

Comment:
Are we sure of this behavior? For me, if the indicator represents the status of the query, and it is errored, I'd show the indicator anyway to give as much as possible information to the user.

Reply:
Yes, the previous functionality of when we show the error of the data in a panel covered only the case when we have a message, did not rely on the `data.state` prop at all. This is because the error can be coming from more than one place, i.e. [here](https://github.com/grafana/grafana/blob/polibb/panel-tests/public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx#L295) and [here](https://github.com/grafana/grafana/blob/polibb/panel-tests/public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx#L367). 

We can look into that, if there's a better way, but for now we're covering the functionality as it is before the new designs.

Reply:
Regardless of the  `statusMessage` presence, we should show the status when the loading state is Error. Otherwise user will see errored panel, but will not be aware of that.

Reply:
I understand your point of view, I used to think that's the way to go as well. @torkelo changed the implementation [in a follow-up PR here](https://github.com/grafana/grafana/pull/61595/commits/5f36a3c714a0543faa764f8baa88068952b6850d) (lines 106 and 152), so I looked into it then and I noticed then that this is indeed what the current functionality of showing the error is: [panel header corner error](https://github.com/grafana/grafana/blob/polibb/panel-tests/public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderCorner.tsx#L98)

Reply:
@dprokop @polibb @ivanortegaalba 

I think it's up the user (developer) of PanelChrome to set statusMessage when there is an error, showing an error state with no information/message is not that useful. I think it's much clearer if loadingState only controls the loading animation and streaming state, and statusMessage only controls the statusMessage. I think it will be more complex and confusing if loadingState also could cause a statusMessage to show even when statusMessage is not there / empty. 

thoughts?

Reply:
I think from the point of view of `PanelChrome`, it receives a `loadingState` that can be one of `'NotStarted' | 'Loading' | 'Streaming' | 'Done' | 'Error'`, so currently there's a distinction between 'Not Started', 'Done', and 'Error'. If we ignore 'Error' we are saying that all three have the same response on the UI of PanelChrome.

Moreover, right now only `statusMessage` shows the errored status and `statusMessageOnClick` does not have that power. It seems to me a valid case to have an error and see no immediate message, but to be able to click on it to trigger an action that will show you something more/take you somewhere else. (We have a default message 'Error in plugin', but that doesn't mean everyone would)

Hence, I would allow the user to see _the errored sign_ if `loadingState is 'Error'`, if there is a `statusMessage`, **or** if there is a `statusMessageOnClick`. Or maybe **only** rely on the loadingState and leave for the dev to set up statusMessage and statusMessageOnClick if they have use of them?

Those are my thoughts atm.

Reply:
IMO having a panel in an erroneous state without showing any indication is a bad, confusing UX.

Can't we have a default status message, like "Something went wrong...." to show when the state is error?

Reply:
I like @dprokop idea. IMO:

There is a single status for a panel: `loading | error | streaming | done | etc`. But it is composed of two props: `loadingState` and `statusMessage`.  The status message is contextual information about the current status. I can imagine the future having `LoadingStatus.Warning` and having the status message as well. In this case, the status would be a warning.

So, IMO I wouldn't ignore the LoadingState in any case and default to a generic message if it is not provided.

Reply:
Okay. I wasn't sure a default message on PanelChrome's part is a good idea, but it does make sense in the context that there is an indicator when there's an error, but one wouldn't necessarily know which indicator that is 😊🙏.

@torkelo what do you think?

Reply:
I think everyone is very confused 

Let's have different props control different things, ok? loadingState => controls loading animation, statusMessage controls statusMessge. To me this seems so much easier from the consumer's (developer) point of view for this component

Reply:
I think it would be fine if the panel showed a read border or something via LoadingState.Error, then the prop would control that aspect (loading border animation, streaming state, error border). but let's not conflate it with the statusMessage. I think that should only be set with the statusMessage prop. Very easy for the consumer of the component to then control when and how the statusMessage is shown.

Reply:
Okay 🙌, statusMessage adds a status icon with a message on the header, and if we one day have an indicator that the loadingState holds 'Error' it will be a separate concern. 

With this in mind, this PR has no need for further changes.